src/controllers/collisions.rs

```diff
@@ -7,15 +11,27 @@ const SCORE_PER_ENEMY: u32 = 10;
 pub struct CollisionsController;
 
 impl CollisionsController {
-    pub fn handle_collisions(state: &mut GameState) {
-        CollisionsController::handle_bullet_collisions(state);
-        CollisionsController::handle_player_collisions(state);
+    pub fn handle_collisions(app: &mut ApplicationState, ctx: &mut Context) {
+        // Don't handle collisions if the player is dead
+        if app.game_state.world.player.is_dead { return; }
+
+        CollisionsController::handle_bullet_collisions(&mut app.game_state);
+        CollisionsController::handle_player_collisions(&mut app.game_state);
+        
+        // If the player died then we set a timeout (3 seconds) after which a game over message
+        // will appear, and the user will be able to restart.
+        if app.game_state.world.player.is_dead {
+            // TODO: use closures here ?
+            fn handler(app: &mut ApplicationState) { app.game_state.game_over(); };
```

Comment:
@aochagavia I wanted to use `let handler = || app.game_state.game_over();` here instead of declaring a new function - but I couldn't get it to compile no matter what I tried. I guess my understanding of Rust's closures isn't that great 😅